.gitlab/kernel_version_testing.yml

```diff
@@ -0,0 +1,71 @@
+  script:
+    # make sure that pahole is available
+    - pahole --version
+    # Fetch relevant kernel commit
+    - mkdir $LINUX_SRC_DIR
+    - pushd $(dirname $LINUX_SRC_DIR)
+    - MAJOR=$(echo $TARGET_TAG | cut -d '.' -f 1)
+    - VERSION=$(echo $TARGET_TAG | cut -d 'v' -f 2)
+    - TARGET_DIR=$(echo $LINUX_SRC_DIR | rev | cut -d '/' -f 1 | rev)
+    - TARGET_ARCHIVE=$TARGET_DIR.tar.gz
+    - wget -q -c "https://mirrors.edge.kernel.org/pub/linux/kernel/$MAJOR.x/linux-$VERSION.tar.gz" -O $TARGET_ARCHIVE
+    - tar -xf $TARGET_ARCHIVE -C $TARGET_DIR --strip-components=1
+    - popd
+    # Copy extra config to source directory
+    - cp /extra.config-$PARENT_EXTRA_KCONFIG_VERSION $LINUX_SRC_DIR/extra.config
+    # Build kernel
+    - make -C $LINUX_SRC_DIR clean
+    - make -C $LINUX_SRC_DIR defconfig
+    - make -C $LINUX_SRC_DIR kvm_guest.config
+    - tee -a < $LINUX_SRC_DIR/extra.config $LINUX_SRC_DIR/.config
+    - make -C $LINUX_SRC_DIR olddefconfig
+    - make -j$(nproc) -C $LINUX_SRC_DIR bindeb-pkg LOCALVERSION=-ddvm
+    - mkdir kernel-$TARGET_TAG.$ARCH.pkg
+    - cp $LINUX_SRC_DIR/arch/$ARCH/boot/bzImage kernel-$TARGET_TAG.$ARCH.pkg/
+    - find $LINUX_SRC_DIR/.. -name linux-headers-$(echo $TARGET_TAG | tr -d 'v')* -type f | xargs -i cp {} kernel-$TARGET_TAG.$ARCH.pkg/
+    - find $LINUX_SRC_DIR/.. -name linux-image-$(echo $TARGET_TAG | tr -d 'v')* -type f | grep -Fv dbg | xargs -i cp {} kernel-$TARGET_TAG.$ARCH.pkg/
+    - tar -czvf kernel-$TARGET_TAG.$ARCH.pkg.tar.gz kernel-$TARGET_TAG.$ARCH.pkg
+  variables:
+    TAGS_COMMITS_FILE: /tags-commits
+    ARCH: x86
+  artifacts:
+    paths:
+      - kernel-$TARGET_TAG.$ARCH.pkg.tar.gz
+
+build_kernels_x64:
+  extends: .build_kernels
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/kernel-version-testing_x64:v$PARENT_PIPELINE_ID
+  tags: [ "runner:main" ]
+  parallel:
+    matrix:
+      - TARGET_TAG: ["v5.0","v5.1","v5.2","v5.3","v5.4","v5.5","v5.6","v5.7","v5.8","v5.9","v5.10","v5.11","v5.12","v5.13","v5.14","v5.15","v5.16","v5.17","v5.18","v5.19","v4.4","v4.5","v4.6","v4.7","v4.8","v4.9","v4.10","v4.11","v4.12","v4.13","v4.14","v4.15","v4.16","v4.17","v4.18","v4.19","v4.20"]
+
+upload_kernel_packages_x64:
+  stage: upload_kernels
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/docker:20.10-py3
+  tags: [ "runner:main" ]
+  needs: [ "build_kernels_x64" ]
+  script:
+    - KARCH=$(echo $ARCH | cut -d '_' -f 1)
+    - find . -name "kernel-*.${KARCH}.pkg.tar.gz" -type f | rev | cut -d '/' -f 1 | rev > packages.txt
+    - DIR=$(dirname $(head -n 1 packages.txt))
+    - cat packages.txt
+    - cd $DIR; tar -cf $KERNEL_PKG_NAME -T packages.txt
+    - sha256sum $KERNEL_PKG_NAME
+    - $S3_CP_CMD $KERNEL_PKG_NAME $S3_DD_AGENT_OMNIBUS_KERNEL_VERSION_TESTING_URI/kernel-packages-$ARCH.tar --grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers
+  variables:
+    KERNEL_PKG_NAME: kernel-packages-$ARCH.tar
+    ARCH: "x86_64"
```

Comment:
There are no arm64 versions of these jobs (though we do build a `kernel-version-testing_arm64` image), is this intended?

Reply:
I am trying to resolve some issues compiling arm kernels. Once resolved I will add an arm64 version of this job as well.